src/compiler/commandLineParser.ts

```diff
@@ -374,10 +374,10 @@ namespace ts {
       * Read tsconfig.json file
       * @param fileName The path to the config file
       */
-    export function readConfigFile(fileName: string): { config?: any; error?: Diagnostic }  {
+    export function readConfigFile(fileName: string, system: System): { config?: any; error?: Diagnostic }  {

```

Comment:
I would use `readFile: (path: string) => string` instead of full-blown `sys` as an optional argument. For consumers that does not have `sys` it will be simpler then supplying implementation of `System` where everything but `readFile` throws `Error("Unsupported")` or something like that.